net-utils/src/lib.rs

```diff
@@ -147,46 +143,90 @@ pub fn verify_reachable_ports(
         return ok;
     }
 
-    for _udp_retries in 0..5 {
-        // Wait for a datagram to arrive at each UDP port
-        for udp_socket in udp_sockets {
-            let port = udp_socket.local_addr().unwrap().port();
-            let udp_socket = udp_socket.try_clone().expect("Unable to clone udp socket");
-            let (sender, receiver) = channel();
-            std::thread::spawn(move || {
-                let mut buf = [0; 1];
-                debug!("Waiting for incoming datagram on udp/{}", port);
-                match udp_socket.recv(&mut buf) {
-                    Ok(_) => sender
-                        .send(())
-                        .unwrap_or_else(|err| warn!("send failure: {}", err)),
-                    Err(err) => warn!("udp recv failure: {}", err),
-                }
-            });
-            match receiver.recv_timeout(Duration::from_secs(5)) {
```

Comment:
Well, I dunno why `channel()` is needed.. I've simplified it just by `socket.set_read_timeout()`....